tests/unit/multiclass/tree/RandomForest_unittest.cc

```diff
@@ -279,3 +283,165 @@ TEST(RandomForest,classify_non_nominal_test)
+	EXPECT_EQ(0.0, values_vector[4]);
+	EXPECT_EQ(1.0, values_vector[5]);
+	EXPECT_EQ(1.0, values_vector[6]);
+	EXPECT_EQ(1.0, values_vector[7]);
+	EXPECT_EQ(1.0, values_vector[8]);
+	EXPECT_EQ(1.0, values_vector[9]);
+
+	SG_UNREF(result);
+}
+
+TEST(RandomForest, score_compare_sklearn_toydata)
+{
+	// Comparison with sklearn's RandomForest probability outputs
+	// https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/ensemble/tests/test_voting_classifier.py#L143
+	float64_t data_A[] = {-1.1, -1.5, -1.2, -1.4, -3.4, -2.2, 1.1, 1.2};
+
+	SGMatrix<float64_t> data(data_A, 2, 4, false);
+
+	CDenseFeatures<float64_t>* features_train =
+	    new CDenseFeatures<float64_t>(data);
+
+	float64_t labels[] = {0.0, 0.0, 1.0, 1.0};
+	SGVector<float64_t> lab(labels, 4);
+	CMulticlassLabels* labels_train = new CMulticlassLabels(lab);
+
+	float64_t expected_probabilities[] = {0.2, 0.2, 0.8, 0.7};
+
+	CRandomForest* c = new CRandomForest(features_train, labels_train, 10, 2);
+	SGVector<bool> ft = SGVector<bool>(2);
+	ft[0] = false;
+	ft[1] = false;
+	c->set_feature_types(ft);
+
+	CMeanRule* mr = new CMeanRule();
+	c->set_combination_rule(mr);
+	c->train(features_train);
+
+	auto result = c->apply_binary(features_train);
+	SGVector<float64_t> res_vector = result->get_labels();
+	SGVector<float64_t> values_vector = result->get_values();
+
+	EXPECT_EQ(-1.0, res_vector[0]);
+	EXPECT_EQ(-1.0, res_vector[1]);
+	EXPECT_EQ(+1.0, res_vector[2]);
+	EXPECT_EQ(+1.0, res_vector[3]);
+
+	for (auto i = 0; i < 4; ++i)
+	{
+		EXPECT_NEAR(expected_probabilities[i], values_vector[i], 11e-1);
+	}
+
+	SG_UNREF(result);
+}
+
+TEST(RandomForest, score_random_labels)
+{
+	// Test data generated using script
+	// https://gist.github.com/olinguyen/d93cd4222b404d26dc859f100d700be6
+	// Generates data with random labels where trees don't all agree
+	sg_rand->set_seed(42);
+
+	float64_t data_A[] = {5, 7, 5, 9, 9, 6, 9, 5, 8, 7,
+	                      3, 1, 4, 1, 1, 2, 1, 0, 2, 0};
+
+	SGMatrix<float64_t> data(data_A, 2, 10, false);
+
+	CDenseFeatures<float64_t>* features_train =
+	    new CDenseFeatures<float64_t>(data);
+
+	float64_t labels[] = {0, 0, 0, 1, 0, 1, 0, 1, 0, 1};
+	SGVector<float64_t> lab(labels, 10);
+	CMulticlassLabels* labels_train = new CMulticlassLabels(lab);
+
+	float64_t test_A[] = {9, 2, 5, 8, 7, 3, 3, 1, 7, 6};
+
+	SGMatrix<float64_t> test_data(test_A, 2, 5, false);
+
+	CDenseFeatures<float64_t>* features_test =
+	    new CDenseFeatures<float64_t>(test_data);
+
+	CRandomForest* c = new CRandomForest(features_train, labels_train, 10, 2);
+	SGVector<bool> ft = SGVector<bool>(2);
+	ft[0] = false;
+	ft[1] = false;
+	c->set_feature_types(ft);
+
+	CMeanRule* mr = new CMeanRule();
+	c->set_combination_rule(mr);
+	c->train(features_train);
+	auto result = c->apply_binary(features_test);
+	SGVector<float64_t> res_vector = result->get_labels();
+	SGVector<float64_t> values_vector = result->get_values();
+
+	EXPECT_NEAR(0.7, values_vector[0], 1e-1);
```

Comment:
thats better!
But I suggest in fact to

 * generate the data randomly (CMath::randn_double)
 * generate the labels randomly, sign(randn())
 * increase the dataset size
 * assert for 0.5